log/logger.go

```diff
@@ -356,20 +316,22 @@ var defaultLogFileSpawners = []logFileSpawner{
 			AttributeKeyRepository,
 			AttributeKeyTask,
 		},
-		FilePath: func(attributes map[AttributeKey]string) string {
+		Spawn: func(attributes map[AttributeKey]string) (slog.Handler, error) {
 			resultPath := attributes[AttributeKeyResultPath]
 			modelID := attributes[AttributeKeyModel]
 			languageID := attributes[AttributeKeyLanguage]
 			repositoryName := attributes[AttributeKeyRepository]
 			taskIdentifier := attributes[AttributeKeyTask]
 
-			return filepath.Join(resultPath, taskIdentifier, CleanModelNameForFileSystem(modelID), languageID, repositoryName, "evaluation.log")
+			file, err := openLogFile(filepath.Join(resultPath, taskIdentifier, CleanModelNameForFileSystem(modelID), languageID, repositoryName, "evaluation.log"))
+			if err != nil {
+				return nil, err
+			}
+
+			return slog.NewJSONHandler(file, nil), nil
```

Comment:
Every `handlerSpawner` has now the this redundant block. What about doing this:

```
filePath := filepath.Join(...)

return newJSONHandlerForFile(filePath)
```

Reply:
I am introducing a helper later on anyways. So maybe I just stuff it in there?